Refresh the whole docs site

MTFT-Games opened the pull request:
General:
- [x] Enable auto table of contents
- [x] Enable auto2top
- [ ] Proofread all content to ensure it is readable and grammatically correct.

Cleanup:
- [x] Remove random unnecessary stuff
- [ ] Fill or remove any empty pages
- [x] Remove sections that are now redundant
- [ ] Restructure sidebar / site
- [x] Remove old library page now that it gets pulled directly from the repo

Styles:
- [x] Replace current logo with the one that has text
- [x] Set the theme back to being from a cdn
- [x] Improve the "Edit on Github" button

Content:
- [x] Add a readme to the docs site
- [x] Make the site pull readme files from the other devcade repos
- [x] Add info to the home page
- [x] Update contribution guide
- [ ] Review all content to ensure accuracy and sufficient detail
- [x] Add the docs site readme to the docs site
- [ ] Confirm all links are working
- [x] Add links to all repos (beyond the edit button)
- [ ] Add links between pages where appropriate
- [ ] Add more content to `first-game.md`
- [ ] Add content to `cabinet.md` (cad files, parts list, cost breakdown, links, etc)
- [ ] Add content to `gamepad.md` (parts list, cost breakdown, links, etc)
- [ ] Add pages explaining how to make a devcade game in other languages/programs/engines (at least the ones that people have made games with so far)
- [ ] Add instructions/documentation for using cargo devcade

Devcade Install Guide (@esoccoli):
- [ ] Remove all references to CSH specific aspects of the setup (ie: csh.rit.edu hostname, mentions of RTPs/Devcade Admins, etc)
- [ ] Add info about configuring openbox
- [ ] Add more headers and subheaders to make it easier to follow.
- [ ] Maybe restructure and/or reword certain things to make it easier to follow
- [ ] Make sure that the onboard will actually compile and run on this setup
  - [ ] If it doesn't, figure out why and add instructions for what needs to be done so it can run

MTFT-Games committed:
Add a readme for the repo

MTFT-Games committed:
Cleaned up index.html
Title got overriden and did nothing
description was not used and seems to do nothing.

MTFT-Games committed:
Set the theme back to a cdn link and killed shortcut icon
The site doesn't work without the cdn for docsify scripts so loading them from cdn but not the theme just doesn't make sense.
The shortcut icon link seemed to do nothing so I removed it because I can.

MTFT-Games committed:
Added pulling in of other readmes
Now it will pull from the readme in each repo. edit on github fix to come soon
Im gonna sleep now

MTFT-Games committed:
Swap edit on github plugin for custom function
This is to allow it to work properly with docs pulled from the other repos.
Also rigt aligned it like the original plugin did.

MTFT-Games committed:
Replaced shortcut icon
Apparently it did did do things, I just didn't notice
Probably caching being a bitch

MTFT-Games committed:
Cleanup
- Removed a blenk line
- Removed the local dark style

MTFT-Games committed:
Added editing instructions to readme

MTFT-Games committed:
Swap logo
Thinner one with text I think looks better

MTFT-Games committed:
Add table of contents in files

MTFT-Games committed:
Removed library page
It was a copy of the readme from the library repo

MTFT-Games committed:
Refreshed the home page
Added a quick start section
Added a note on editing

MTFT-Games committed:
Added auto2top

MTFT-Games committed:
Make it darker

MTFT-Games committed:
Quick typo fix in readme

MTFT-Games committed:
Typo fix on flatpakify

MTFT-Games committed:
Create contributing guide

MTFT-Games committed:
Remove now redundant sections
Now that readmes are pulled directly from repos, the outdated static versions here can go

MTFT-Games committed:
Add style guides to contributing
The agreed upon standards were found and placed int he contributing file.

pikachu0542 committed:
Updated the dcu install guide and reorganized the order of items on the sidebar

pikachu0542 commented:
Specific changes to make
- [ ] Add more detail to the "Making your first devcade game" page
- [x] Update content of the onboard readme (it is comically out of date)
- [ ] Fix the link to (and possibly update content of) the "Using the Game Template" page
- [ ] Add more detail to the hardware overview page
- [ ] Add content to or remove the "Cabinet" page
- [ ] Add content to or remove the "Gamepad" page
- [ ] Add headings to the "Installing DCU" page
- [ ] Update all links in the "Internals" section
- [ ] Possibly add more detail to the "API" page
- [ ] Possibly add more detail to the "Monogame Game Template" page
- [ ] Possibly add more detail to the "Flatpakify" page

pikachu0542 committed:
Updated the using the game template page, and a couple very small misc changes

MTFT-Games committed:
Add heading, summary, and hardware back to install guide

MTFT-Games commented:
@esoccoli Some notes on the new install guide:
In addition to adding headers as you mentioned above, I would like to see it formatted more like a walk through or guide like the old one was.
Similarly, I've been trying to format the docs site with general information for anyone that might in the future want to make their own Devcade. To this point I would like to avoid specifying details specific to our instance of Devcade such as the dcu.csh.rit.edu hostname or referencing RTPs/Devcade Admins. These belong better on our wiki I think. 
Do we need to specify mary's fork of flatpak still? or have the changes we need trickled down to the standard package yet?
We include an openbox config with the onboard, we could just add the volume control config to that... Or maybe as a separate PR.

MTFT-Games committed:
Re-add contributing guide to sidebar

MTFT-Games committed:
Change up home page
Now more of a 'heres where to start' instead of just a list of some of whats already on the sidebar

MTFT-Games committed:
Move repo readmes under new how it works

MTFT-Games committed:
Fix link to new How it works

MTFT-Games commented:
@esoccoli Thoughts on not having the using-template page and instead just including that info in the readme of the template itself?

MTFT-Games committed:
Filled out the How it Works section

pikachu0542 commented:
Id be fine with that. Most of the content is kinda just a repeat of what is in the readme of the template anyway

pikachu0542 commented:
What I will say is that i can't guarantee that the onboard will actually launch after you finish following the instructions in the install guide. I was trying to test that, and it kept giving me errors, and I couldn't figure out why, and didn't really have the time to look into it too much.

pikachu0542 committed:
Misc spelling and grammar fixes, also rewrote a paragraph on one of the pages to make it more readable

pikachu0542 committed:
Added docsify to the package.json, updated readme

MTFT-Games committed:
Spellcheck how-it-works.md

MTFT-Games committed:
various added info

MTFT-Games committed:
Move how it works to an overview sub page

MTFT-Games committed:
Fill out most of cabinet.md

MTFT-Games committed:
Fixed lack of newline after bullet point

pikachu0542 committed:
Renamed the 'getting started' section to 'making games'

pikachu0542 committed:
Moved the control diagram from Making games to gamepad

pikachu0542 committed:
Split up contributing guide into the general contributing.md and a separate file for style guides

pikachu0542 committed:
Expanded the gamepad page

pikachu0542 committed:
Improved the first game page

pikachu0542 committed:
Did some reorganization and de-redundancy-ifying the docs

pikachu0542 committed:
pushing local changes

AlexanderCarlisi committed:
Add wiring section to gamepad.md

pikachu0542 committed:
Merge pull request #14 from AlexanderCarlisi/docs-refresh

Add wiring section to gamepad.md

pikachu0542 commented:
Merging this now bc there is a lot of good stuff, and I want to get what we have into main so it is easier to find, and will continue updating stuff as we go

pikachu0542 committed:
Merge into main

pikachu0542 reviewed (approved):
lgtm
